src/js/crud.js

```diff
@@ -0,0 +1,153 @@
+// Check if authorized
+if (sessionStorage.getItem("auth") !== "granted") {
+  alert("Unauthorized access!");
+  window.location.href = "login.html";
+}
+
+import { getMenuItemsData, getApiUrl } from './fetch.js';
+
+// Predefined tags for products (can be replaced with an API call if necessary)
+const predefinedTags = [
+  "bread", "artisan", "savory", "vegan", "special!", 
+  "pastry", "buttery", "breakfast", "sweet", "dessert",
+  "chocolate", "fruit", "snack", "cake", "citrus"
+];
+
+// Helper functions
+function highlightError(inputEl) {
+  inputEl.style.borderColor = "red";
+  inputEl.focus();
+}
+
+function clearErrors(inputs) {
+  inputs.forEach(i => i.style.borderColor = "#ccc");
+}
+
+// Populate the Tags dropdown with predefined tags
+function populateTagsDropdown() {
+  const tagsSelect = document.getElementById("tags");
+  
+  predefinedTags.forEach(tag => {
+    const option = document.createElement("option");
+    option.value = tag;
+    option.textContent = tag;
+    tagsSelect.appendChild(option);
+  });
+}
+
+// Fetch and render products
+async function fetchProducts() {
+  const products = await getMenuItemsData();
+  const list = document.querySelector('.product-list');
+  list.innerHTML = "";
+
+  products.forEach(p => {
+    const li = document.createElement("li");
+    li.innerHTML = `
```

Comment:
Ik zou dit niet zo doen. Als je namelijk op de `<button>` nog een eventlistener wilt zetten, kan de browser problemen veroorzaken. https://www.geeksforgeeks.org/javascript/what-is-the-disadvantage-of-using-innerhtml-in-javascript/